src/frequenz/dispatch/_actors_service.py

```diff
@@ -0,0 +1,225 @@
+# License: All rights reserved
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Helper class to manage actors based on dispatches."""
+
+import asyncio
+import logging
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from frequenz.channels import Broadcast, Receiver
+from frequenz.client.dispatch.types import TargetComponents
+from frequenz.sdk.actor import Actor, BackgroundService
+
+from ._dispatch import Dispatch
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class DispatchUpdate:
```

Comment:
As I said in the other PR, I would probably call this `DispatchInfo` (or maybe `DispatchDetails`?), as it will also represent the initial dispatch information, not just an "update". `initial_dispatch: DispatchUpdate` sounds a bit weird to me. `new_with_dispatch(initial_details: DispatchDetails, ...)` or `new_with_dispatch(initial_info: DispatchInfo, ...)` sounds better to me.